proposed/tensor/type-and-memory-comparison.md

```diff
@@ -0,0 +1,52 @@
+
+| | libTorch | ONNX | Numpy | TensorFlow | Julia |
+| --- | --- | --- | --- | --- | --- |
+| **Types Supported** | | | | | |
+|  |  |  |  |  |
+| byte | X | X | | X | X | X |
+| sbyte | X | X | X | X | X |
+| ushort | X | X | X | X | X |
+| short | X | X | X | X | X |
+| int | X | X | X | X | X |
+| uint | X | X | X | X | X |
+| long | X | X | X | X | X |
+| ulong | X | X | X | X | X |
+| bool (byte) | X | X | X | X | X |
+| string (bytes) |  | X | X | X
```

Comment:
@NiklasGustafsson @tannergooding is using tensors with string elements common? It seems like it might be difficult to reconcile with a tensor type that has a generic number constraint.

Reply:
Same concern applies to a few other types in this list.